### PR TITLE
hydra: plugins_path and advanced config

### PR DESCRIPTION
**Pull request by dberenbaum**
Partially addresses two different issues:

- #5094 - adds docs for https://github.com/iterative/dvc/pull/10240 but not https://github.com/iterative/dvc/pull/9783
- #4670 - adds hydra-related content but not dvc.yaml features like templating

### CODE REVIEW

**Comment by dberenbaum** [comment]
@AlexandreKempf It would be great to get your help with some of the docs issues. PTAL and let me know if you have thoughts or questions.

File: content/docs/user-guide/project-structure/configuration.md [hunk 1]
```diff
@@ -258,12 +258,19 @@ Composition].
...
+- `hydra.plugins_path` - location of the parent directory of `hydra_plugins`,
+  where Hydra will automatically discover [plugins]. Defaults to the root of the
+  DVC repository.
 
 [config composition]:
   https://hydra.cc/docs/tutorials/basic/your_first_app/composition/
 [config groups]:
   https://hydra.cc/docs/tutorials/basic/your_first_app/config_groups/
 [defaults list]: https://hydra.cc/docs/tutorials/basic/your_first_app/defaults/
+[config module]:
```
Comment (shcheklein) [inline content/docs/user-guide/project-structure/configuration.md:270]: is it referenced anywhere? just to double check

**Review by shcheklein — APPROVED** [review]
just a few things to double check / cleanup. Thanks @dberenbaum !

File: content/docs/user-guide/experiment-management/hydra-composition.md [hunk 2]
```diff
@@ -221,4 +222,102 @@ Stage 'train' didn't change, skipping
...
+separation between config and code can help debug because the entire config
+generated by Hydra gets written to `params.yaml` before the experiment starts.
+You can also reuse `params.yaml` across multiple scripts in different stages of
+a DVC pipeline.
+
+## Advanced Hydra config
+
+You can configure how DVC works with Hydra.
+
+By default, DVC will look for Hydra [config groups] in a `conf` directory, but
```
[the remark this answers is not in the document]
Comment (dberenbaum) [reply content/docs/user-guide/experiment-management/hydra-composition.md:277]: Yep, it's already linked in the doc.

File: content/docs/user-guide/project-structure/configuration.md (hunk 1 above)
Comment (dberenbaum) [reply content/docs/user-guide/project-structure/configuration.md:270]: Good catch on this one.

File: content/docs/user-guide/experiment-management/hydra-composition.md [hunk 3]
```diff
@@ -221,4 +222,102 @@ Stage 'train' didn't change, skipping
...
+    main()
+```
+
+To convert the same code to use DVC with Hydra composition enabled:
+
+```python
+from omegaconf import OmegaConf
+
+def main() -> None:
+    cfg = OmegaConf.load("params.yaml")
```
Comment (AlexandreKempf) [inline content/docs/user-guide/experiment-management/hydra-composition.md:259]: Shouldn't the yaml file named `"config.yaml"` in that case?

File: content/docs/user-guide/experiment-management/hydra-composition.md [hunk 4]
```diff
@@ -221,4 +222,102 @@ Stage 'train' didn't change, skipping
...
+The final `params.yaml` will look like:
+
+```yaml
+dir: raw/data
+relpath: dataset.csv
+fullpath: raw/data/dataset.csv
+```
+
+[plugins]:
+  https://hydra.cc/docs/advanced/plugins/develop/#automatic-plugin-discovery-process
```
Comment (AlexandreKempf) [inline content/docs/user-guide/experiment-management/hydra-composition.md:321]: I'm probably wrong, but you don't seem to use this link anywhere in the doc. Is that normal?

**Review by AlexandreKempf — CHANGES_REQUESTED** [review]
I'm not 100% familiar with the subject (I have used hydra and DVC, but never together). Your additional doc looks clear on how to use them together. Maybe one or two concerns before I approve it :smiley:

File: content/docs/user-guide/experiment-management/hydra-composition.md (hunk 4 above)
Comment (dberenbaum) [reply content/docs/user-guide/experiment-management/hydra-composition.md:321]: Yes, good catch! I added a link above to reference this.

**Pull request merged**